shared/src/main/scala/scopt/ORunner.scala

```diff
@@ -324,9 +324,15 @@ private[scopt] object ORunner {
             i += option.tokensToRead(i, args) - 1
           } // if
         case None =>
+          def isShortOpt(arg: String): Boolean =
+            arg.startsWith("-") && (arg.length == 1 || ! {
+              val c = arg.charAt(1)
```

Comment:
Would this work for `foo -123foo`? I think checking just the first letter is asking for future trouble.

Reply:
Would `-123foo` be an option anyone would be using? The reason I am only checking the first character is that otherwise you would probably be having to parse more complicated strings such as `1.0e-2` etc. and determine whether they are valid numbers or not.

Reply:
Or put differently, should `-1e2` be a valid option?

Reply:
I think treating `-1e2` as `Read[Double]` is a nice-to-have, but not necessary.

Reply:
Currently you _can_ have `1e2` and it will come out as `200`. I know checking just the first character is not particularly "safe", but if you think about the whole construction, you would actually have to check against the defined options first and then backtrack (so you would have to be able to handle the meaning of `-1e2` based on the available options and pending arguments), and that will make the parsing much more complicated. I'm not aware of command line tools in the wild that use `-1` or `-2` as options, but I might be wrong.

Reply:
[ssh](https://linux.die.net/man/1/ssh) and rsync comes to mind.

> -1        Forces ssh to try protocol version 1 only.

Reply:
They would you have to rewrite the parser, because you cannot backtrack from `handleShortOptions`, it mutates the state on the way and can not fail safely...

Reply:
Another more quick solution would be to optionally _enable_ this parsing, so by default it would not accept negative numbers, but you can make it do so. It could be as simple as maing `isShortOpt` a protected method, and so one can enable that behaviour with an override, and the current behaviour will remain the default.

Reply:
`-1` as option seems like a pathological corner case, so I am ok with not handling it too.